compiler/src/main/java/com/squareup/anvil/compiler/codegen/PsiUtils.kt

```diff
@@ -278,6 +278,13 @@ internal fun PsiElement.requireFqName(
         module.findClassOrTypeAlias(importFqName, classReference)?.let { return it.fqNameSafe }
       }
 
+  // Check if it's a named import
+  containingKtFile.importDirectives
+    .firstOrNull { classReference == it.importPath?.importedName?.asString() }?.importedFqName
+    ?.let {
+      module.resolveClassByFqName(it, FROM_BACKEND)?.let { return it.fqNameSafe }
```

Comment:
That seems unnecessary. Can't you just return `it` immediately without resolving? You already have the fully qualified name.

Reply:
Makes sense, if it matches the reference of the class I can just return it 🤦